### PR TITLE
Hide useless comments

### PR DESCRIPTION
**Pull request by bfred-it**
Closes https://github.com/sindresorhus/refined-github/issues/1478

Testable on https://github.com/hayaku/hayaku/issues/303

This is how the comments reappear after clicking "Undo"

### CODE REVIEW

File: source/features/hide-useless-comments.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+export default function () {
+	if (!pageDetect.isIssue() && !pageDetect.isPR()) {
+		return;
+	}
+
+	let uselessCount = 0;
+	for (const commentText of select.all('.comment-body > p:only-child')) {
+		// Find useless comments
+		if (!/^(\+[120]+!?|👍|🙏)$/.test(commentText.textContent.trim())) {
```
Comment (sindresorhus) [inline source/features/hide-useless-comments.js]: Can you support `-1` and `👎` too?

Comment (bfred-it) [reply source/features/hide-useless-comments.js]: Surr

**Comment by yakov116** [comment]
Can the note be on top and on bottom of the page?

**Comment by bfred-it** [comment]
Why?

File: source/features/hide-useless-comments.js (hunk 1 above)
Comment (busches) [reply source/features/hide-useless-comments.js]: @bfred-it don't you mean 👍  ;)

Comment (bfred-it) [reply source/features/hide-useless-comments.js]: :+1;

**Comment by yakov116** [comment]
When unhiding more comments it does not hide them. Is that done purposely?

**Comment by bfred-it** [comment]
Yeah it does not watch the page for new comments be uncollapsed comments, I thought that if you click that and nothing happens it would seem confusing/broken. So it’s best to leave the useless comments visible.

**Comment by sindresorhus** [comment]
Looks good. Just needs to be added to the readme.

**Comment by yakov116** [comment]
@bfred-it https://github.com/sindresorhus/ama/issues/363#issuecomment-193159839 

>Hides 👍 👎 comments and shows their count in the sidebar
>>When there's only a few of them they might just be "okay, that works" rather than "I vote for this feature"

Flash back 😄

**Comment by bfred-it** [comment]
You found that 😂 

Glad that the reasoning still changes. Conversational +1’s aren’t removed (however it doesn’t consider the count)

**Comment by Mottie** [comment]
> Yeah it does not watch the page for new comments be uncollapsed comments

It doesn't appear to be checking for useless comments when the "xxx hidden items Load more..." button is clicked; like on [this page](https://github.com/isaacs/github/issues/215).

And I don't think it catches comments like "+1!" (exclamation points). There's a lot of them in that thread.

**Comment by bfred-it** [comment]
> It doesn't appear to be checking for useless comments when the "xxx hidden items Load more..." button is clicked; like on [this page](https://github.com/isaacs/github/issues/215).

Refer to https://github.com/sindresorhus/refined-github/pull/1526#issuecomment-421507010

> And I don't think it catches comments like "+1!" (exclamation points). There's a lot of them in that thread.

`+111111!!!!!!` comments are caught (`/^([+-]\d+!*|👍|🙏|👎|👌)+$/`)

screen shot 2018-09-18 at 08 55 14

**Comment by darkred** [comment]
The "xxx useless comments were hidden" counter appears to be greater than the comments total count:
in the test URL from the OP, https://github.com/hayaku/hayaku/issues/303 ,
it's `123 useless comments were hidden` (in both Chrome 69 And Firefox 62)
but in the first post it's shown that there are "102 comments" in total.

**Comment by bfred-it** [comment]
That’s because the regex was updated after that screenshot and now it catches more comments 👌

**Comment by darkred** [comment]
_(Follow-up to my comment above)_

If you visit the test page, https://github.com/hayaku/hayaku/issues/303 now, 
the "xxx hidden items Load more..." button now has 45: 
*<sup>(in my comment above, three days ago, it had 25)</sup>*


the page still has the same comments count, 102:


and the extension currently shows a different, still incorrect count,"86 useless comments were hidden"

while, if I'm not mistaken, the actually hidden comments count is now 44.


PS. Chrome 69, extension version 18.9.19.1710.

**Comment by bfred-it** [comment]
I think you’re right. The miscount is because GitHub sends the same comment twice in the code to enable some features. To fix it, the first selector needs to include `.unminimized-comment` or something like that.

**Comment by bfred-it** [comment]
Show an example of “longer comment” hidden

**Comment by hubertgrzeskowiak** [comment]
With hidden comments:
screen shot 2018-09-26 at 2 29 43 pm

After undoing the effects of this plugin:
screen shot 2018-09-26 at 2 29 55 pm

**Pull request merged**